bot/exts/core/help.py

```diff
@@ -41,14 +41,18 @@ class HelpQueryNotFound(ValueError):
     """
     Raised when a HelpSession Query doesn't match a command or cog.
 
-    Contains the custom attribute of ``possible_matches``.
-    Instances of this object contain a dictionary of any command(s) that were close to matching the
-    query, where keys are the possible matched command names and values are the likeness match scores.
+    Params:
+    possible_matches: list of similar command names.
+    parent_command: parent command of an invalid subcommand. Only available when an invalid subcommand
+                    has been passed.
     """
 
-    def __init__(self, arg: str, possible_matches: dict = None):
+    def __init__(
+            self, arg: str, possible_matches: list[str] = None, *, parent_command: Command = None
```

Comment:
Missing `Optional` or `| None` typing.
```suggestion
            self, arg: str, possible_matches: list[str] | None = None, *, parent_command: Command | None = None
```

Reply:
Good catch, though we need to use `Optional` from the `typing` module since the project is running on 3.9 and not 3.10.